Fix/timelines api migration

apfelfabrik opened the pull request:
Wraps up https://github.com/opf/openproject/pull/527:

> roll-up branch for migrating the rest of the API-views to RABL (given, that we actually make things faster...)

This one also restores some timelines functionality. ~~Unfortunately, Timelines are still missing the capability to fetch users trough their new API, which is why this PR explicitly leaves [`#2577`](https://www.openproject.org/issues/2577) unfixed.~~ Thanks, @nkenneweg.

hschink commented on app/assets/javascripts/timelines.js:
```diff
@@ -721,7 +722,7 @@ Timeline = {
 
         this.augmentPlanningElementsWithHistoricalData();
-        this.augmentPlanningElementsWithProjectAndParentAndChildInformation();
+        this.augmentPlanningElementsWithAllKindsOfStuff();
```
Eventually, this will lead to maintenance hell! :fire: Let it in for fun! :laughing:

apfelfabrik committed:
Restores changelog.

apfelfabrik committed:
Merge remote-tracking branch 'opf/feature/rails3' into fix/timelines-api-migration

Conflicts:
	doc/CHANGELOG.md

apfelfabrik committed:
Merge remote-tracking branch 'opf/fix/timelines-api-migration' into fix/timelines-api-migration

Conflicts:
	doc/CHANGELOG.md

apfelfabrik commented:
Merged w/ other changes once more, Changelog is prepared to merge fine w/ release 23.

hschink commented:
@apfelfabrik, @nkenneweg I'm sorry but tests (cucumber and specs) fail both on Travis and the internal CI. I propose to fix those tests before merging.

apfelfabrik commented:
Nooooooooooooooooooooo!

apfelfabrik committed:
Removes obsolete spec.

apfelfabrik committed:
Fixes expectations for planning element controller spec.

apfelfabrik committed:
Fixes timeline comparison cukes.

Status didn't have workflows assigned to them in the role and type that
was used, so timelines didn't receive the right set of them.

nkenneweg commented on doc/CHANGELOG.md:
```diff
@@ -38,9 +46,9 @@ See doc/COPYRIGHT.rdoc for more details.
-* `#2221` [Accessibility] enhance keyboard shortcuts
 * `#2026` 404 error when letters are entered in category Work Package
 * `#2129` Repository: clicking on file results in 500
+* `#2221` [Accessibility] enhance keyboard shortcuts
```
Der steht oben auch nochmal. Komisch.

tilt committed:
Fix typo in api/v2/planning_element_journals#index

tilt commented:
We could sometime update the routes
-      resources :planning_element_journals
-      resources :statuses
-      resources :planning_element_journals, only: [:index, :create]
-      resources :statuses, only: [:index, :show]
  ...
